### PR TITLE
update for change to MIM(PS) prefix in DOID

### LINKED ISSUE #560: DO mappings: unable to import (`OMIM`(`PS`) --> `MIM`
## Overview
We now have less DO mappings (+2177 -13,705) as of the last build (#556).

## Probable cause of issue
Following Nico's suspicion and Trish's [initial investigation](https://github.com/monarch-initiative/mondo-ingest/pull/556#discussion_r1630269317 ), I looked at the `components/doid.owl` for this build, and compared it to an older one. Indeed, they did do these changes.

```owl
<!-- Previously: OMIM -->
<owl:Class rdf:about="http://purl.obolibrary.org/obo/DOID_0050331">
    <oboInOwl:hasDbXref>MIM:149730</oboInOwl:hasDbXref>

<!-- Previously: OMIMPS -->
<owl:Class rdf:about="http://purl.obolibrary.org/obo/DOID_0050328">
    <oboInOwl:hasDbXref>MIM:PS275200</oboInOwl:hasDbXref>
```

The also previously had some references directly to OMIM URIs, e.g. `https://omim.org/entry/...`, but those are removed. I only see CURIE references now.

## Possible solutions
1. Replace our `prefix_map`s to account for `MIM` (https://github.com/monarch-initiative/omim/issues/111)
2. Any other solution?

## Additional info
Context and more information:
- https://github.com/monarch-initiative/mondo-ingest/pull/556#discussion_r1630269317 

[before and after - doid.sssom.tsv.zip](https://github.com/user-attachments/files/15754225/before.and.after.-.doid.sssom.tsv.zip)

DO had a new release on 5/29/2024. That release seems to be when they switched prefix maps (e.g. --> MIM).

### PR DESCRIPTION
**Pull request by twhetzel**
Closes #560

## Overview

This PR:
Updates the SPARQL queries used in the `components/doid.owl` goal in `mondo-ingest.Makefile` since the Disease Ontology (DOID) now uses the prefixes MIM and MIM:PS for classes.
```
<owl:Class rdf:about="http://purl.obolibrary.org/obo/DOID_0070431">
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/DOID_0060308"/>
        <obo:IAO_0000115 xml:lang="en">An autosomal recessive intellectual developmental disorder characterized by hyperphosphatasia and intellectual disability. Distinctive facial features including hypertelorism, long palpebral fissures, a nose with a broad bridge and a rounded tip, downturned corners of the mouth, and a thin upper lip are also often observed.</obo:IAO_0000115>
        <oboInOwl:hasDbXref>MESH:C565495</oboInOwl:hasDbXref>
        **<oboInOwl:hasDbXref>MIM:PS239300</oboInOwl:hasDbXref>**
        <oboInOwl:hasDbXref>ORDO:247262</oboInOwl:hasDbXref>
        <oboInOwl:hasDbXref>SNOMEDCT_US_2023_03_01:33982008</oboInOwl:hasDbXref>
        <oboInOwl:hasDbXref>UMLS_CUI:C1855923</oboInOwl:hasDbXref>
        <oboInOwl:hasExactSynonym xml:lang="en">HPMRS</oboInOwl:hasExactSynonym>
        <oboInOwl:hasExactSynonym xml:lang="en">Mabry disease</oboInOwl:hasExactSynonym>
        <oboInOwl:hasExactSynonym xml:lang="en">Mabry syndrome</oboInOwl:hasExactSynonym>
        <oboInOwl:hasExactSynonym xml:lang="en">hyperphosphatasia with mental retardation syndrome</oboInOwl:hasExactSynonym>
        <oboInOwl:hasOBONamespace>disease_ontology</oboInOwl:hasOBONamespace>
        <oboInOwl:id>DOID:0070431</oboInOwl:id>
        <oboInOwl:inSubset rdf:resource="http://purl.obolibrary.org/obo/doid#DO_rare_slim"/>
        <rdfs:label xml:lang="en">hyperphosphatasia with impaired intellectual development syndrome</rdfs:label>
        **<skos:exactMatch>MIM:PS239300</skos:exactMatch>**
        <skos:exactMatch>ORDO:247262</skos:exactMatch>
        <skos:exactMatch>UMLS_CUI:C1855923</skos:exactMatch>
    </owl:Class>
```

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [x] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [ ] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
- [x] --> In progress, will be pushed as branch `issue-560-BUILD-2` and run using `sh run.sh make build-mondo-ingest -B`
   
See:
- #569

### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [ ] Yes

### CODE REVIEW

File: src/sparql/fix_make_omim_exact.ru [hunk 1]
```diff
@@ -23,12 +23,12 @@ WHERE
...
-    FILTER( STRSTARTS(str(?value2), "OMIM"))
+    FILTER( STRSTARTS(str(?value2), "MIM"))
     FILTER(?value!=?value2)
   }
   
-  FILTER( STRSTARTS(str(?value), "OMIM"))
+  FILTER( STRSTARTS(str(?value), "MIM"))
   FILTER( !isBlank(?cls) && STRSTARTS(str(?cls), "http://purl.obolibrary.org/obo/DOID_"))
-  BIND(IRI(REPLACE(REPLACE(STR(?value), "OMIMPS:", "https://omim.org/phenotypicSeries/PS"), "OMIM:", "https://omim.org/entry/")) as ?iri)
+  BIND(IRI(REPLACE(REPLACE(STR(?value), "MIMPS:", "https://omim.org/phenotypicSeries/PS"), "MIM:", "https://omim.org/entry/")) as ?iri)
```
Comment (joeflack4) [inline src/sparql/fix_make_omim_exact.ru:32]: Just dropping some initial thoughts here.

Looking at `component-download-doid.owl`, I see that references like
```owl
        <oboInOwl:hasDbXref>OMIM:PS267700</oboInOwl:hasDbXref>
```

have changed to
```owl
        <oboInOwl:hasDbXref>MIM:PS267700</oboInOwl:hasDbXref>
```

Don't have time at the moment to think deeply about this; I'm not sure if this solution works or not. I can think more about it, but we can also use it and see what happens.

Just dropping that this is related:
- https://github.com/monarch-initiative/omim/issues/111

I suppose that in tandem with this, we have to also make updates to the `omim` repo.

**Review by matentzn — APPROVED** [review]
I think this looks fine!

Note that if there was (I dont think there is) a process that relies on the xrefs to be formatted OMIM:123 and OMIM:PS123, then this PR will remove that possibility.

I am fine with waving it through and trying, but you could if you like

- [x] 1. Optional: Add a step to normalise MIM:123 to OMIM:123
- [x] 2. Optional: Instead of normalising MIM:PS123 to MIMPS:123 normalise it to OMIMPS:123

But this is optional

**Comment by twhetzel** [comment]
Given the [successful Build](https://github.com/monarch-initiative/mondo-ingest/pull/569) from this code update, making this an Active PR now.

File: src/sparql/fix_make_omim_exact.ru (hunk 1 above)
Comment (twhetzel) [reply src/sparql/fix_make_omim_exact.ru]: Resolving comment - we realized these should just be reverted back to OMIMPS

**Review by matentzn — APPROVED** [review]
Great, thank you!

File: src/sparql/fix_omimps.ru [hunk 2]
```diff
@@ -24,7 +24,7 @@ WHERE
            owl:annotatedTarget ?value ;
            ?p ?v .
   }
-  FILTER(STRSTARTS(str(?value), "OMIM:PS"))
+  FILTER(STRSTARTS(str(?value), "MIM:PS"))
   FILTER (isIRI(?entity))
-  BIND(REPLACE(?value, "OMIM:PS", "OMIMPS:", "i") AS ?value_fixed)
+  BIND(REPLACE(?value, "MIM:PS", "OMIMPS:", "i") AS ?value_fixed)
```
Comment (joeflack4) [inline src/sparql/fix_omimps.ru:29]: ## `fix_omimps.ru`
These changes LGTM!

File: src/sparql/fix_mim.ru [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  OPTIONAL {
+    ?ax a owl:Axiom ;
+           owl:annotatedSource ?entity ;
+           owl:annotatedProperty xref: ;
+           owl:annotatedTarget ?value ;
+           ?p ?v .
+  }
+  FILTER(STRSTARTS(str(?value), "MIM"))
+  FILTER (isIRI(?entity))
+  BIND(REPLACE(?value, "MIM", "OMIM", "i") AS ?value_fixed)
```
Comment (joeflack4) [inline src/sparql/fix_mim.ru:29]: ## `fix_mim.ru`
New file, I see. LGTM!

File: src/ontology/mondo-ingest.Makefile [hunk 4]
```diff
@@ -107,6 +107,7 @@ $(COMPONENTSDIR)/doid.owl: $(TMPDIR)/doid_relevant_signature.txt | component-dow
 		remove -T $(TMPDIR)/doid_relevant_signature.txt --select complement --select "classes individuals" --trim false \
 		query \
 			--update ../sparql/fix_omimps.ru \
+			--update ../sparql/fix_mim.ru \
```
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:110]: ## SPARQL query order
The order of these queries looks correct to me!

Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:110]: ## Approve
I wasn't sure about the last version of these changes from the other day, but what you have in this PR immediately strikes me as clear and correct! And the [build](https://github.com/monarch-initiative/mondo-ingest/pull/569) results also seem to corroborate.

**Review by joeflack4 — APPROVED** [review]

**Pull request merged**